internal/client.go

```diff
@@ -1003,6 +1027,32 @@ func DialCloudOperationsClient(ctx context.Context, options CloudOperationsClien
 	}, nil
 }
 
+// PrepareUpdateWorkflowOperation returns an UpdateWorkflowOperation that can be used to perform Update-with-Start,
+// or an error in case the update operation is invalid.
+func PrepareUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
+	input, err := createUpdateWorkflowInput(options)
+	if err != nil {
+		return nil, err
+	}
+	return &UpdateWorkflowOperation{input: input}, nil
+}
+
+// Get returns a WorkflowUpdateHandle, or an error if the workflow run is invalid.
+func (op *UpdateWorkflowOperation) Get(run WorkflowRun) (WorkflowUpdateHandle, error) {
+	if run == nil {
+		return nil, errors.New("invalid workflow run: is nil")
+	}
+	if runImpl, ok := run.(*workflowRunImpl); ok {
+		if runImpl.startRequest != op.startRequest {
+			return nil, errors.New("invalid workflow run: does not belong to update operation")
+		}
+	}
```

Comment:
Open to other ideas; this was the best way I could think of to ensure the operation and workflow run match.